File: InputTestCode/Kotlin/KotlinBasics/12_Collections.kt
// Collections

fun inCollection(items: Array<String>): String {
    var str = "Nothing to say."

    when {
        "orange" in items -> str = "juicy"
        "apple" in items -> str = "apple is fine too"
    }

    return str
}

fun collectionSimpleLambda(list: Array<String>): List<String> {
    return list.filter { it == "win" }
}

fun collectionLambdaExpression(list: Array<String>): MutableList<String> {
    val filteredList = mutableListOf<String>()

    list
        .filter { it.startsWith("a") }
        .sortedBy { it }
        .map { it.toUpperCase() }
        .forEach { filteredList.add(it) }

    return filteredList
}

File: InputTestCode/Swift/KotlinBasics/12_Collections.swift
// Collections

func inCollection(items: [String]) -> String {
    var str = "Nothing to say."

    if(items.contains("orange")) {
        str = "juicy"
    } else if(items.contains("apple")) {
        str = "apple is fine too"
    }

    return str
}

func collectionSimpleLambda(list: [String]) -> [String] {
    return list.filter { $0 == "win" }
}

func collectionLambdaExpression(list: [String]) -> [String] {
    var filteredList = [String]()

    list
        .filter({ $0.starts(with: "a") })
        .sorted()
        .map({ $0.uppercased() })
        .forEach({ filteredList.append($0) })

        return filteredList
}